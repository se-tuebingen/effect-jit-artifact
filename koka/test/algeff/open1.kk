// test lifting of cps functions
fun myid() : ((x:a) -> e a) {
  return fn(x){ x }
}

val hs : forall<e> list<int -> e int> = [myid()]

fun bar() {
  head(hs).default(id)(10)
}

fun main() {
  bar().show.println
}