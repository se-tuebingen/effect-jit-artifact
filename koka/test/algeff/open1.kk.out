10
 
algeff/open1/bar: () -> int
algeff/open1/hs: forall<e> list<(int) -> e int>
algeff/open1/main: () -> console ()
algeff/open1/myid: forall<a,e> () -> ((x : a) -> e a)